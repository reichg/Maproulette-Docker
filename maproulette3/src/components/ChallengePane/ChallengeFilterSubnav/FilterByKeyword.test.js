import React from 'react'
import { FilterByKeyword } from './FilterByKeyword'
import { cloneDeep as _cloneDeep } from 'lodash'

const propsFixture = {
  user: {
    id: 11,
    savedChallenges: [],
  },
  challenges: [
    {
      id: 309,
      name: "Challenge 309",
      blurb: "Challenge 309 blurb",
      description: "Challenge 309 description",
      parent: {
        displayName: "foo",
      }
    },
    {
      id: 311,
      name: "Challenge 311",
      blurb: "Challenge 311 blurb",
      description: "Challenge 311 description",
      parent: {
        displayName: "bar",
      }
    }
  ],
  searchFilters: {
    difficulty: 1,
    keywords: 'Challenge'
  }
}

let basicProps = null

beforeEach(() => {
  basicProps = _cloneDeep(propsFixture)
  basicProps.setKeywordFilter = jest.fn()
  basicProps.removeSearchFilters = jest.fn()
  basicProps.intl = {formatMessage: jest.fn()}
})

test("it renders with props as expected", () => {
  const wrapper = shallow(
    <FilterByKeyword {...basicProps} />
  )

  expect(wrapper).toMatchSnapshot()
})

test("it calls setKeywordFilter if an onChange occurs with a value", () => {
  const wrapper = shallow(
    <FilterByKeyword {...basicProps} />
  )

  wrapper.instance().updateFilter({value: 'water'})
  expect(basicProps.setKeywordFilter).toBeCalledWith(["natural", "water"])
})

test("it calls removeSearchFilters if an onChange occurs with a null value", () => {
  const wrapper = shallow(
    <FilterByKeyword {...basicProps} />
  )

  wrapper.instance().updateFilter({value: null})
  expect(basicProps.removeSearchFilters).toBeCalledWith(['keywords'])
})
