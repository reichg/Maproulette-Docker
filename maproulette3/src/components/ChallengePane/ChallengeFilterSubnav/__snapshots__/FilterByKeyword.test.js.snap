// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders with props as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterByKeyword
    challenges={
      Array [
        Object {
          "blurb": "Challenge 309 blurb",
          "description": "Challenge 309 description",
          "id": 309,
          "name": "Challenge 309",
          "parent": Object {
            "displayName": "foo",
          },
        },
        Object {
          "blurb": "Challenge 311 blurb",
          "description": "Challenge 311 description",
          "id": 311,
          "name": "Challenge 311",
          "parent": Object {
            "displayName": "bar",
          },
        },
      ]
    }
    intl={
      Object {
        "formatMessage": [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "defaultMessage": "Roads / Pedestrian / Cycleways",
                "id": "Challenge.keywords.navigation",
              },
            ],
            Array [
              Object {
                "defaultMessage": "Water",
                "id": "Challenge.keywords.water",
              },
            ],
            Array [
              Object {
                "defaultMessage": "Points / Areas of Interest",
                "id": "Challenge.keywords.pointsOfInterest",
              },
            ],
            Array [
              Object {
                "defaultMessage": "Buildings",
                "id": "Challenge.keywords.buildings",
              },
            ],
            Array [
              Object {
                "defaultMessage": "Land Use / Administrative Boundaries",
                "id": "Challenge.keywords.landUse",
              },
            ],
            Array [
              Object {
                "defaultMessage": "Transit",
                "id": "Challenge.keywords.transit",
              },
            ],
            Array [
              Object {
                "defaultMessage": "Other",
                "id": "Challenge.keywords.other",
              },
            ],
            Array [
              Object {
                "defaultMessage": "Anything",
                "id": "Challenge.keywords.any",
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
      }
    }
    removeSearchFilters={[MockFunction]}
    searchFilters={
      Object {
        "difficulty": 1,
        "keywords": "Challenge",
      }
    }
    setKeywordFilter={[MockFunction]}
    user={
      Object {
        "id": 11,
        "savedChallenges": Array [],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "button": <ButtonFilter
        type={
          <FormattedMessage
            defaultMessage="Work on"
            id="ChallengeFilterSubnav.filter.keyword.label"
            values={Object {}}
          />
        }
      />,
      "children": <ol
        className="mr-list-dropdown mr-list-dropdown--ruled"
      >
        <li>
          <a
            onClick={[Function]}
          />
        </li>
        <li>
          <a
            onClick={[Function]}
          />
        </li>
        <li>
          <a
            onClick={[Function]}
          />
        </li>
        <li>
          <a
            onClick={[Function]}
          />
        </li>
        <li>
          <a
            onClick={[Function]}
          />
        </li>
        <li>
          <a
            onClick={[Function]}
          />
        </li>
        <li>
          <a
            onClick={[Function]}
          />
        </li>
        <li>
          <OtherKeywordsOption
            clearSearch={[Function]}
            placeholder=""
            searchQuery={
              Object {
                "query": "C",
              }
            }
            setSearch={[Function]}
          />
        </li>
      </ol>,
      "className": "mr-dropdown--flush xl:mr-border-l xl:mr-border-white-10 mr-p-6 mr-pl-0 xl:mr-pl-6",
      "close": [Function],
      "isVisible": false,
      "toggleVisible": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <a
              onClick={[Function]}
            />
          </li>,
          <li>
            <a
              onClick={[Function]}
            />
          </li>,
          <li>
            <a
              onClick={[Function]}
            />
          </li>,
          <li>
            <a
              onClick={[Function]}
            />
          </li>,
          <li>
            <a
              onClick={[Function]}
            />
          </li>,
          <li>
            <a
              onClick={[Function]}
            />
          </li>,
          <li>
            <a
              onClick={[Function]}
            />
          </li>,
          <li>
            <OtherKeywordsOption
              clearSearch={[Function]}
              placeholder=""
              searchQuery={
                Object {
                  "query": "C",
                }
              }
              setSearch={[Function]}
            />
          </li>,
        ],
        "className": "mr-list-dropdown mr-list-dropdown--ruled",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "any",
          "nodeType": "host",
          "props": Object {
            "children": <a
              onClick={[Function]}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "navigation",
          "nodeType": "host",
          "props": Object {
            "children": <a
              onClick={[Function]}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "water",
          "nodeType": "host",
          "props": Object {
            "children": <a
              onClick={[Function]}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "pointsOfInterest",
          "nodeType": "host",
          "props": Object {
            "children": <a
              onClick={[Function]}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "buildings",
          "nodeType": "host",
          "props": Object {
            "children": <a
              onClick={[Function]}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "landUse",
          "nodeType": "host",
          "props": Object {
            "children": <a
              onClick={[Function]}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "transit",
          "nodeType": "host",
          "props": Object {
            "children": <a
              onClick={[Function]}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "other",
          "nodeType": "host",
          "props": Object {
            "children": <OtherKeywordsOption
              clearSearch={[Function]}
              placeholder=""
              searchQuery={
                Object {
                  "query": "C",
                }
              }
              setSearch={[Function]}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "clearSearch": [Function],
              "placeholder": "",
              "searchQuery": Object {
                "query": "C",
              },
              "setSearch": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ol",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "button": <ButtonFilter
          type={
            <FormattedMessage
              defaultMessage="Work on"
              id="ChallengeFilterSubnav.filter.keyword.label"
              values={Object {}}
            />
          }
        />,
        "children": <ol
          className="mr-list-dropdown mr-list-dropdown--ruled"
        >
          <li>
            <a
              onClick={[Function]}
            />
          </li>
          <li>
            <a
              onClick={[Function]}
            />
          </li>
          <li>
            <a
              onClick={[Function]}
            />
          </li>
          <li>
            <a
              onClick={[Function]}
            />
          </li>
          <li>
            <a
              onClick={[Function]}
            />
          </li>
          <li>
            <a
              onClick={[Function]}
            />
          </li>
          <li>
            <a
              onClick={[Function]}
            />
          </li>
          <li>
            <OtherKeywordsOption
              clearSearch={[Function]}
              placeholder=""
              searchQuery={
                Object {
                  "query": "C",
                }
              }
              setSearch={[Function]}
            />
          </li>
        </ol>,
        "className": "mr-dropdown--flush xl:mr-border-l xl:mr-border-white-10 mr-p-6 mr-pl-0 xl:mr-pl-6",
        "close": [Function],
        "isVisible": false,
        "toggleVisible": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <a
                onClick={[Function]}
              />
            </li>,
            <li>
              <a
                onClick={[Function]}
              />
            </li>,
            <li>
              <a
                onClick={[Function]}
              />
            </li>,
            <li>
              <a
                onClick={[Function]}
              />
            </li>,
            <li>
              <a
                onClick={[Function]}
              />
            </li>,
            <li>
              <a
                onClick={[Function]}
              />
            </li>,
            <li>
              <a
                onClick={[Function]}
              />
            </li>,
            <li>
              <OtherKeywordsOption
                clearSearch={[Function]}
                placeholder=""
                searchQuery={
                  Object {
                    "query": "C",
                  }
                }
                setSearch={[Function]}
              />
            </li>,
          ],
          "className": "mr-list-dropdown mr-list-dropdown--ruled",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "any",
            "nodeType": "host",
            "props": Object {
              "children": <a
                onClick={[Function]}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "navigation",
            "nodeType": "host",
            "props": Object {
              "children": <a
                onClick={[Function]}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "water",
            "nodeType": "host",
            "props": Object {
              "children": <a
                onClick={[Function]}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "pointsOfInterest",
            "nodeType": "host",
            "props": Object {
              "children": <a
                onClick={[Function]}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "buildings",
            "nodeType": "host",
            "props": Object {
              "children": <a
                onClick={[Function]}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "landUse",
            "nodeType": "host",
            "props": Object {
              "children": <a
                onClick={[Function]}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "transit",
            "nodeType": "host",
            "props": Object {
              "children": <a
                onClick={[Function]}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "other",
            "nodeType": "host",
            "props": Object {
              "children": <OtherKeywordsOption
                clearSearch={[Function]}
                placeholder=""
                searchQuery={
                  Object {
                    "query": "C",
                  }
                }
                setSearch={[Function]}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "clearSearch": [Function],
                "placeholder": "",
                "searchQuery": Object {
                  "query": "C",
                },
                "setSearch": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ol",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
